### PR TITLE
CRITICAL BUG FIXES + Walkthrough updates and minor fixes imported from old repo

### PR DESCRIPTION
**Pull request by starwarigami**
Production system is currently unusable due to various 500 errors following change to postgres backend. These fixes address the worst of them that I have found. It also includes some of the feature updates to walkthroughs, and fixes for issues with the f-puzzles import

### CODE REVIEW

**Comment by SorenHolstHansen** [comment]
Great! However as far as I can see, this PR doesn't change any code, only linting?

File: src/stores/sudokuStore/interactionHandlers.ts [hunk 1]
```diff
@@ -60,6 +61,9 @@ export const defaultHandleMouseEnter: MouseEnterHandler = ({ cell, mouseDown })
 
 /** Default action to run when user moves around with the arrow buttons */
 export const defaultHandleArrows: ArrowHandler = ({ k }) => {
+  //do not accept keyboard input when any modal controls are open
+  if (hasOpenModals()) return;
```
Comment (SorenHolstHansen) [inline src/stores/sudokuStore/interactionHandlers.ts:65]: We should also make this check the places where we use a custom arrow handlers

**Review by SorenHolstHansen — CHANGES_REQUESTED** [review]
Thanks a lot for this, really appreciate it! Have a few comments though

File: src/models/User.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export const UserValidator = z.object({
     ),
   /** The email of the user */
   email: z.string().email(),
-  password: z.string().min(10).max(64),
+  password: z.string().min(10).max(256),
```
[the remark this answers is not in the document]
Comment (starwarigami) [reply src/models/User.ts:19]: it gives a 500 error clicking on a puzzle to view it. This was only intended to be a temporary patch to get the production system up and running. The query is pulling back the hashed password from the db, which is longer than the 64 char limit, throwing an internal error and failing to load the puzzle

File: src/stores/sudokuStore/interactionHandlers.ts (hunk 1 above)
Comment (starwarigami) [reply src/stores/sudokuStore/interactionHandlers.ts:65]: done

File: src/models/User.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (SorenHolstHansen) [reply src/models/User.ts:19]: Arrh, cool. I just merged a fix for this, so you can revert these changes

File: src/stores/sudokuStore/interactionHandlers.ts (hunk 1 above)
Comment (SorenHolstHansen) [reply src/stores/sudokuStore/interactionHandlers.ts:65]: Great!

**Comment by starwarigami** [comment]
I'm back at work today, so won't be able to make any further changes till
this evening

**Review by SorenHolstHansen — APPROVED** [review]

**Pull request merged**